src/ecs/resources/input/input_handler.rs

```diff
@@ -358,9 +309,10 @@ impl InputHandler {
     // Pressed actions has been omitted as the function is a little difficult to write and I
     // can't think of a use case for it.
 
-    /// Checks if the given button is currently pressed.
+    /// Checks if the given action is currently pressed.
     pub fn action_is_pressed<T: AsRef<str>>(&self, action: T) -> Option<bool> {
```

Comment:
@torkleyy I implemented a modified version of the ButtonState idea, please review.